### PR TITLE
Update k8s tests and libraries to v1.24.0

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

```release-note
Add support for Kubernetes v1.24.0
```

*Note for reviewers*: This PR was tested against kernel 4.9 and k8s 1.24. The tests that are currently failing are known as they are also failing for other Kubernetes versions and thus not being k8s specific. Since it's currently not possible to update the net-next VM I'm proposing to change running the net-next w/ k8s 1.24 to running kernel 4.9 with k8s 1.24.

### CODE REVIEW

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

**Comment by aanm** [comment]
/test-1.24-4.9

**Comment by aanm** [comment]
/test-1.24-4.9

**Comment by aanm** [comment]
/test-1.24-4.9

**Review by rolinh — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by nebril — APPROVED** [review]

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 1]
```diff
@@ -54,17 +54,17 @@ pipeline {
...
             steps {
-                // retrieve k8s and kernel versions from gh comment, then from job parameter, default to 1.23 for k8s, 419 for kernel
+                // retrieve k8s and kernel versions from gh comment, then from job parameter, default to 1.24 for k8s, 419 for kernel
                 script {
                     flags = env.ghprbCommentBody?.replace("\\", "")
                     env.K8S_VERSION = sh script: '''
                         if [ "${ghprbCommentBody}" != "" ]; then
                             python3 ${TESTDIR}/get-gh-comment-info.py ''' + flags + ''' --retrieve="k8s_version" | \
-                            sed "s/^$/${JobK8sVersion:-1.23}/" | \
+                            sed "s/^$/${JobK8sVersion:-1.24}/" | \
```
Comment (nebril) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile:63]: I don't think any jenkins job uses the default value RN, we still need to configure new jobs with `1.24`.

**Comment by aanm** [comment]
/test-1.24-4.9

**Comment by aanm** [comment]
The k8s tests failed for https://github.com/cilium/cilium/issues/17270

**Comment by aanm** [comment]
/test-1.24-4.19

**Pull request merged**